### PR TITLE
OHRM5X-2486: Bump FE testing libraries to latest major version

### PR DESCRIPTION
**Pull request by devishke-orange**
PR contains:
- OHRM5X-2486: [FE] Update libraries with security alerts

What's changed:
* Set `yarn` to version 4.1.0
* Updated all `test/functional` dependencies to the latest major version using `yarn upgrade-interactive`
* Migrated cypress tests
* Introduced new endpoints to functionalTestingPlugin to speed up cypress tests

### CODE REVIEW

File: src/test/functional/cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
+const {defineConfig} = require('cypress');
```
Comment (RajithaKumara) [inline src/test/functional/cypress.config.js:1]: add copyright header

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -94,9 +94,9 @@ jobs:
         run: |
           docker exec os_dev_php80 bash -c 'php src/test/functional/tools/prepare.php'
           cd html/src/test/functional
-          sed -i 's~http://php80/orangehrm/web/index.php~http://php80/web/index.php~g' cypress.json
-          sed -i 's~false~true~g' cypress.json
-          sed -i 's~6000~10000~g' cypress.json
+          sed -i 's~http://php80/orangehrm/web/index.php~http://php80/web/index.php~g' cypress.config.js
+          sed -i 's~false~true~g' cypress.config.js
```
Comment (RajithaKumara) [inline .github/workflows/cypress.yml:98]:
```suggestion
          sed -i 's~video: false~video: true~g' cypress.config.js
```
Have to check further

File: .github/workflows/cypress.yml [hunk 3]
```diff
@@ -94,9 +94,9 @@ jobs:
...
+          sed -i 's~6000~10000~g' cypress.config.js
```
Comment (RajithaKumara) [inline .github/workflows/cypress.yml:99]:
```suggestion
          sed -i 's~defaultCommandTimeout: 6000~defaultCommandTimeout: 10000~g' cypress.config.js
```

File: src/test/functional/cypress.config.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+/// <reference types="cypress" />
+const axios = require('axios');
+
+module.exports = defineConfig({
+  defaultCommandTimeout: 6000,
+  video: false,
+  scrollBehavior: 'top',
+  experimentalInteractiveRunEvents: true,
```
Comment (RajithaKumara) [inline src/test/functional/cypress.config.js:10]: check whether this feature still experimental

File: src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js [hunk 5]
```diff
@@ -42,17 +41,29 @@ describe('Admin - Employment Status API ', function () {
 
   describe('With `emp-statuses` snapshot', function () {
     before(function () {
-      cy.task('db:reset');
+      cy.task('db:truncate', {tables: ['EmploymentStatus']});
```
Comment (RajithaKumara) [inline src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js:44]: can you revert these to `db:reset` as we discuss. will improve it in a separate PR

File: src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js [hunk 6]
```diff
@@ -42,17 +41,29 @@ describe('Admin - Employment Status API ', function () {
...
+        cy.apiLogin(admin);
+      });
       cy.request('POST', empStatusAPI, {
         name: this.strings.chars50.text,
       }).then(function () {
-        cy.task('db:snapshot', {name: 'emp-statuses'});
+        cy.task('db:snapshotSpecific', {
+          savepointName: 'emp-statuses',
+          tables: ['ohrm_employment_status'],
+        });
```
Comment (RajithaKumara) [inline src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js:54]: may be revert this also

File: src/test/functional/tools/plugins/orangehrmFunctionalTestingPlugin/Service/DatabaseBackupService.php [hunk 7]
```diff
@@ -241,4 +241,89 @@ private function deleteAllSavepoints(): bool
...
+    public function createSpecificSavepoint(string $savepointName, array $tableNames): void
+    {
+        $this->_createSpecificSavepoint($this->genSavepointCacheKeyPrefix($savepointName), $tableNames);
+    }
+
+    /**
+     * @param string $savepointCacheKeyPrefix
+     * @param string[] $tableNames
+     */
+    private function _createSpecificSavepoint(string $savepointCacheKeyPrefix, array $tableNames): void
```
Comment (RajithaKumara) [inline src/test/functional/tools/plugins/orangehrmFunctionalTestingPlugin/Service/DatabaseBackupService.php:258]: let's discuss these offline

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/test/functional/cypress.config.js (hunk 4 above)
Comment (devishke-orange) [reply src/test/functional/cypress.config.js:10]: yes

https://docs.cypress.io/guides/references/experiments#Configuration

File: src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js (hunk 6 above)
Comment (devishke-orange) [reply src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js]: reverted

File: src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js (hunk 5 above)
Comment (devishke-orange) [reply src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js]: reverted

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (devishke-orange) [reply .github/workflows/cypress.yml]: changed to env variables

Reference: https://docs.cypress.io/guides/guides/environment-variables#Overriding-environment-variables

File: src/test/functional/cypress.config.js (hunk 1 above)
Comment (devishke-orange) [reply src/test/functional/cypress.config.js:18]: added

File: src/test/functional/cypress.config.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+  e2e: {
+    baseUrl: 'http://php80/orangehrm/web/index.php',
```
Comment (Super-Chama) [inline src/test/functional/cypress.config.js:29]: Will this be a problem in CICD ?

File: src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js [hunk 9]
```diff
@@ -26,7 +26,6 @@ describe('Admin - Employment Status API ', function () {
 
   beforeEach(function () {
     cy.fixture('chars').as('strings');
-    Cypress.Cookies.preserveOnce('orangehrm', '_orangehrm');
```
Comment (Super-Chama) [inline src/test/functional/cypress/e2e/integration/api/admin/empStatus_API.cy.js:29]: I think we added this to remember the session will it break?

**Review by Super-Chama — COMMENTED** [review]
Added some observations.. maybe not relevant.

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**